src/file.rs

```diff
@@ -66,13 +75,14 @@ pub fn madvise_random(_map: &mut memmap2::MmapMut) {}
 
 #[cfg(not(windows))]
 fn mmap(file: &std::fs::File, len: usize) -> Result<memmap2::MmapMut> {
+	// TODO pass len not 0 and limit reserve to max file size
 	#[cfg(not(test))]
 	const RESERVE_ADDRESS_SPACE: usize = 1024 * 1024 * 1024; // 1 Gb
 														 // Use a different value for tests to work around docker limits on the test machine.
 	#[cfg(test)]
 	const RESERVE_ADDRESS_SPACE: usize = 64 * 1024 * 1024; // 64 Mb
 
-	let map_len = len + RESERVE_ADDRESS_SPACE;
+	let map_len = len + RESERVE_ADDRESS_SPACE; // TODO should be max??
```

Comment:
`RESERVE_ADDRESS_SPACE` is supposed to be additional reserved space. I.e. if `RESERVE_ADDRESS_SPACE` is 1024Mb and we try to map 1025Mb it should reserve additional space up to 2048Mb so that when next time we try to map 1026Mb it could use the same space.

Reply:
if `max_len` is set, it should be used as max map size instead of `RESERVE_ADDRESS_SPACE`